build.zig

```diff
@@ -1,187 +1,93 @@
-        qemu,
+    const run_step = b.step("run", "Run with qemu");
+    const qemu_bin = if (target.getArch() == builtin.Arch.i386) "qemu-system-i386" else unreachable;
+    const qemu_cmd = b.addSystemCommand([_][]const u8{
+        qemu_bin,
         "-cdrom",
         iso_path,
         "-boot",
         "d",
         "-serial",
         "stdio",
-    }) catch unreachable;
+    });
     if (debug)
-        qemu_flags.appendSlice([][]const u8{
-            "-s",
-            "-S",
-        }) catch unreachable;
+        qemu_cmd.addArgs([_][]const u8{ "-s", "-S" });
     if (rt_test)
-        qemu_flags.appendSlice([][]const u8{
-            "-display",
-            "none"
-        }) catch unreachable;
-    const cmd = b.addSystemCommand(qemu_flags.toSlice());
-    step.dependOn(&cmd.step);
-}
+        qemu_cmd.addArgs([_][]const u8{ "-display", "none" });
+    run_step.dependOn(&qemu_cmd.step);
+    qemu_cmd.step.dependOn(&iso_cmd.step);
 
-fn buildISO(b: *Builder, build_path: []const u8, iso_path: []const u8) ArrayList(*Step) {
-    const grub_build_path = concat(b.allocator, build_path, "/iso/boot/") catch unreachable;
-    const iso_dir_path = concat(b.allocator, build_path, "/iso") catch unreachable;
-    const grub_cmd = b.addSystemCommand([][]const u8{ "cp", "-r", "grub", grub_build_path.toSlice() });
-    const iso_cmd = b.addSystemCommand([][]const u8{ "grub-mkrescue", "-o", iso_path, iso_dir_path.toSlice() });
-    var steps = ArrayList(*Step).init(b.allocator);
-    steps.append(&grub_cmd.step) catch unreachable;
-    steps.append(&iso_cmd.step) catch unreachable;
-    return steps;
-}
-
-fn buildLink(b: *Builder, target: builtin.Arch, build_path: []const u8) *Step {
-    const exec = b.addExecutable("pluto.elf", null);
-    const elf_path = concat(b.allocator, build_path, "/iso/boot") catch unreachable;
-    exec.setOutputDir(elf_path.toSlice());
-    exec.setLinkerScriptPath("link.ld");
-    exec.setTarget(target, builtin.Os.freestanding, builtin.Abi.gnu);
-    for (src_files.toSlice()) |file| {
-        var file_obj = concat(b.allocator, build_path, "/") catch unreachable;
-        file_obj.append(file) catch unreachable;
-        file_obj.append(".o") catch unreachable;
-        exec.addObjectFile(file_obj.toSlice());
-        exec.setMainPkgPath(".");
-    }
-    for (src_files_asm.toSlice()) |file| {
-        var file_obj = concat(b.allocator, build_path, "/") catch unreachable;
-        file_obj.append(file) catch unreachable;
-        file_obj.append(".o") catch unreachable;
-        exec.addObjectFile(file_obj.toSlice());
+    const test_step = b.step("test", "Run tests");
+    if (rt_test) {
+        const script = b.addSystemCommand([_][]const u8{ "python3", "test/rt-test.py", "x86", b.zig_exe });
+        test_step.dependOn(&script.step);
+    } else {
+        const unit_tests = b.addTest(main_src);
+        unit_tests.setMainPkgPath(".");
+        unit_tests.addBuildOption(bool, "rt_test", rt_test);
```

Comment:
We shouldn't need this build option for unit tests since they don't require the kernel to be run